src/ui/common/src/components/workflows/nodes/NodeStatus.tsx

```diff
@@ -9,29 +9,25 @@ type Props = {
 };
 
 export const NodeStatus: React.FC<Props> = ({ execState }) => {
-  const statusIcons = [];
-  const runStatus = execState.status.toLowerCase();
+  let statusIcon = null;
+  const runStatus: string = execState.status.toLowerCase();
 
   if (runStatus === ExecutionStatus.Succeeded) {
-    statusIcons.push(<Chip label="Succeeded" color="success" size="small" />);
+    statusIcon = <Chip label="Succeeded" color="success" size="small" />;
```

Comment:
I have no idea why this was an array in the past, since we can only have one status, so i changed it to a singular value instead 🙂

Reply:
This component was for some reason copied from the `WorkflowStatus` component. That one had multiple badges because we were showing a badge for success/failure but also for any warnings. I'm not sure why we have two separate components though -- I created a [task](https://linear.app/aqueducthq/issue/ENG-1630/refactor-workflowstatus-and-nodestatus-components-to-be-a-unified) a while back to merge them.

Reply:
@vsreekanti makes sense. I can tackle that after I get this PR in.

Reply:
I actually don't think we need to spend time on this one immediately. Let's get this PR in and keep pushing on the UI redesign, and we can come back to some of these smaller tasks later on if that makes sense to you.